Source/player.cpp

```diff
@@ -3603,6 +3603,8 @@ void ModifyPlrStr(int p, int l)
 	int max = player.GetMaximumAttributeValue(CharacterAttribute::Strength);
 	if (player._pBaseStr + l > max) {
 		l = max - player._pBaseStr;
+	} else if (player._pBaseStr + l < 0) {

```

Comment:
Looks like a good candidate for `clamp` from https://github.com/diasurgical/devilutionX/blob/master/Source/utils/stdcompat/algorithm.hpp

Reply:
Having a hard time figuring out how to incorporate this

Reply:
Perhaps like this:

```c++
l = clamp(l, 0, max - player._pBaseStr);
```

Reply:
Actually after looking at this, it would prevent stats from being reduced by 1 regardless of base stat